rir/src/compiler/pir/instruction.h

```diff
@@ -1134,17 +1134,79 @@ class FLI(AsTest, 1, Effects() | Effect::Error | Effect::Warn) {
     size_t gvnBase() const override { return tagHash(); }
 };
 
-class FLI(AsInt, 1, Effect::Error) {
+class FLI(ColonInputEffects, 2, Effect::Error) {
   public:
-    bool ceil;
+    explicit ColonInputEffects(Value* lhs, Value* rhs, unsigned srcIdx)
+        : FixedLenInstruction(NativeType::test,
+                              {{PirType::any(), PirType::any()}}, {{lhs, rhs}},
```

Comment:
this instruction expects evaluated values, not promises. so the input types should be `PirType::val()` and not `any()`

Reply:
(same for the other instrs)